test/util/isDomainSpecifiedByUser.spec.ts

```diff
@@ -85,3 +89,184 @@ describe('isDomainSpecifiedByUser', () => {
     expect(isDomainSpecifiedByUser(undefined as never, true, 'number')).toBe(false);
   });
 });
+
+describe('parsing axis domain provided by user', () => {
+  // These are valid domains without any further questions
+  const validCases: MyTestCases = [
```

Comment:
Please focus on the test cases when reviewing. Please let me know if I missed some